### DISCUSSION TITLE
Is there a way to use AgentInstrumentationConfig?

### DISCUSSION THREAD

**Discussion by zerocooldog** [open-telemetry/opentelemetry-java-instrumentation#11909] [Q&A] [answered]
Is there a way to use AgentInstrumentationConfig?

I want to read OpenTelemetry properties while writing an extension, but I'm unsure how to do it. In the source code, I can see AgentInstrumentationConfig and AgentCommanConfig classes, but they are not accessible from the outside. Is there a way to access them?

If not, I am considering implementing a solution on my own.

**Comment by laurit** [comment]
`AgentInstrumentationConfig` is an agent internal class it can change between agent releases. Other than that there are no restrictions using it.

**Comment by zerocooldog** [comment]
@laurit 

I'm aware of the issues that can arise with changes in releases, and I intend to use it with those considerations in mind.

However, I'm encountering an error stating that it cannot be accessed from the IDE tool. 

I'm using IntelliJ IDEA 2024.1.4 (Ultimate Edition).

I'm wondering if this is an issue with the tool, if I need to configure settings such as --add-modules, or if there are any other specific settings I need to apply.

![image](https://github.com/user-attachments/assets/711e5907-bd9b-449f-b4fe-60c8400e41df)

**Reply by laurit** [reply]
Did you try adding the appropriate gradle dependency?

**Reply by zerocooldog** [reply]
@laurit 


I'm providing you with a sample project where the same issue occurs.

I believe I have added all the necessary dependencies, but is there anything else needed? You can check the build.gradle and dependency.gradle source.

https://github.com/zerocooldog/open-telemetry-sample-project

**Reply by laurit** [reply, marked as answer]
error says
```
/Users/ltulmin/dev/open-telemetry-sample-project/src/main/java/org/example/extension/SampleInstrumentationModule.java:23: error: cannot access InstrumentationConfig
        AgentInstrumentationConfig.get().getString("ssss");
```
try also adding the dependency that contains `InstrumentationConfig`

**Reply by zerocooldog** [reply]
@laurit 

I initially thought it depended on another library, so I didn't add it separately. After adding the library directly, everything works fine. Thank you!

![image](https://github.com/user-attachments/assets/1872e7be-0a60-4917-80cf-712081ef2779)
